File: Agentic AI CFG Generation/my-agent/input/Cross Language Equivilance/Point Lies Inside Given N points of a Convex Polygon/PCP.py
def cw(a, b, c):
    p = a[0] * (b[1] - c[1]) + b[0] * (c[1] - a[1]) + c[0] * (a[1] - b[1]);
  
    return p < 0;

def ccw(a, b, c):
    p = a[0] * (b[1] - c[1]) + b[0] * (c[1] - a[1]) + c[0] * (a[1] - b[1]);
  
    return p > 0;

def convexHull(v):

    v.sort();
    
  
    n = len(v);
    if (n <= 3):
        return v;
  
    p1 = v[0];
    p2 = v[n - 1];
  
  
    up = []
    down = [];

    up.append(tuple(p1));
    down.append(p1);

    for i in range(1, n):
        
        if i == n - 1 or (not ccw(p1, v[i], p2)):

            while len(up) > 1 and ccw(up[len(up) - 2], up[len(up) - 1], v[i]): 
                up.pop();
            
  
            up.append(tuple(v[i]));
        
  
        if i == n - 1 and  (not cw(p1, v[i], p2)): 
  
            while (len(down) > 1) and cw(down[len(down) - 2], down[len(down) - 1], v[i]):
                down.pop();
            
            down.append(v[i]);
        
    for i in range(len(down) - 2, -1, -1):
        up.append(tuple(down[i]));
  
    up = set(up)
    up = list(up)
  
    return up;

def isInside( points, query):
    points.append(query);
  
    points = convexHull(points);
  
    for x in points: 
        if x == query:
            return False;
    
    return True;


n = 7;
points = [[1, 1 ], [2, 1 ], [ 3, 1 ], [ 4, 1 ], [ 4, 2 ], [4, 3 ], [ 4, 4 ]];
  
query = [ 3, 2 ];
  
if (isInside(points, query)) :
    print("YES");
    
else :
    print("NO");
    

File: Agentic AI CFG Generation/my-agent/input/Real World Sampling/Point Lies Inside Given N points of a Convex Polygon/PCP.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class PCP {
    static boolean cw(int[] a, int[] b, int[] c) {
        int p = a[0] * (b[1] - c[1]) + b[0] * (c[1] - a[1])
                + c[0] * (a[1] - b[1]);

        return p < 0;
    }

    static boolean ccw(int[] a, int[] b, int[] c) {
        int p = a[0] * (b[1] - c[1]) + b[0] * (c[1] - a[1])
                + c[0] * (a[1] - b[1]);

        return p > 0;
    }

    static int[][] convexHull(int[][] v) {
        Arrays.sort(v, (a, b) -> a[0] - b[0]);

        int n = v.length;
        if (n <= 3) {
            return v;
        }

        int[] p1 = v[0];
        int[] p2 = v[n - 1];

        List<int[]> up = new ArrayList<int[]>();
        List<int[]> down = new ArrayList<int[]>();

        up.add(p1);
        down.add(p1);

        for (int i = 1; i < n; i++) {
            if (i == n - 1 || !ccw(p1, v[i], p2)) {
                while (up.size() > 1
                       && ccw(up.get(up.size() - 2),
                              up.get(up.size() - 1),
                              v[i])) {
                    up.remove(up.size() - 1);
                }

                up.add(v[i]);
            }

            if (i == n - 1 || !cw(p1, v[i], p2)) {
                while (down.size() > 1
                       && cw(down.get(down.size() - 2),
                             down.get(down.size() - 1),
                             v[i])) {
                    down.remove(down.size() - 1);
                }

                down.add(v[i]);
            }
        }

        for (int i = down.size() - 2; i >= 0; i--) {
            up.add(down.get(i));
        }

        return up.toArray(new int[0][]);
    }


    static boolean isInside(int[][] points, int[] query) {
        int[][] points1 = new int[points.length + 1][];
        for (int i = 0; i < points.length; i++)
            points1[i] = points[i];
        points1[points.length] = query;

        points = convexHull(points);

        for (int[] x : points) {
            if (Arrays.equals(x, query))
                return false;
        }

        return true;
    }

    public static void main(String[] args){
        int n = 7;
        int[][] points
            = { new int[] { 1, 1 }, new int[] { 2, 1 },
                new int[] { 3, 1 }, new int[] { 4, 1 },
                new int[] { 4, 2 }, new int[] { 4, 3 },
                new int[] { 4, 4 } };
                
        int[] query = { 3, 2 };

        if (isInside(points, query)) {
            System.out.println("YES");
        }
        else {
            System.out.println("NO");
        }
    }
}